service/controller/resource/monitoring/prometheus/node_count.go

```diff
@@ -0,0 +1,86 @@
+package prometheus
+
+import (
+	"github.com/giantswarm/k8sclient/v7/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"golang.org/x/net/context"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	expv1beta1 "sigs.k8s.io/cluster-api/exp/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/prometheus-meta-operator/v2/service/key"
+)
+
+func getClusterNodeCount(ctx context.Context, k8sClient k8sclient.Interface, cluster metav1.Object) (int, error) {
+	mdCount, err := getClusterNodeCountFromMachineDeployment(ctx, k8sClient, cluster)
+	if err != nil {
+		return 0, microerror.Mask(err)
+	}
+
+	mpCount, err := getClusterNodeCountFromMachinePool(ctx, k8sClient, cluster)
+	if err != nil {
+		return 0, microerror.Mask(err)
+	}
+
+	var nodeCount int32 = 1
```

Comment:
I would go with 4 at a minimum here :)

Reply:
wait, customers can have 0 worker nodes if they want, so 1 might be good 🤔

Reply:
Yes we are only counting worker nodes here. There's no way to get master nodes count on vintage AFAIK. And I did not get how to get this info on CAPI either.

I agree 1 is probably no a good default, but that's what we currently have more or less. But let's test and find a better value here.

Reply:
> I agree 1 is probably no a good default, but that's what we currently have more or less.

I don't think it can relate to "what we currently have".
Today we start with 1G, and that's it.
But maybe the right sizing is like: 500MB / node, so "what we currently have" would be 2?